### DISCUSSION TITLE
Give Users the Option to Control the New Notifications Inbox Refresh Behavior

### DISCUSSION THREAD

**Discussion by RandomDSdevel** [community/community#10313] [Mobile]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;GitHub for iOS v1.48.0 included the following change as documented in its release notes:  

> ⋮ (_Snipped…_)  
>
> - The notifications invox will refresh every time the tab is loaded.  
>
> ⋮ (_Snipped…_)  

This can prove to be problematic or disruptive to end-user workflows; for example:  

1. A user may prefer not to dismiss an issue from their notifications inbox until after they've reviewed both it and a pull request which resolved it.  
1. A user may tap a link to a non-GitHub URI to oprn it in their browser or otherwise navigate away from the GitHub app to do something else, perhaps run a Web search for something if, say, they run into an unfamiliar term and want to look it up.  If this action outside the GitHub app takes long enough, the user may switch back to it only to find that it was suspended by the OS and reset back to the notifications inbox from whichever issue or PR they might've been looking at before.  With the old behavior, getting back to where you were is relatively easy:  you just tap the relevant issue or PR's notification inbox list entry again.  With the new behavior, you have to remember where you were before you switched away from the GitHub app and find it again by navigating through its associated repository's issue or PR list.  

Users should therefore be able to comtrol whether this new behavior triggers or not from the in-app setting page.  

(Feel free to merge this issue with #10276.)

**Comment by Mahdi10458** [comment]
> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;GitHub for iOS v1.48.0 included the following change as documented in its release notes:  
> 
> > ⋮ (_Snipped…_)  
> >
> > - The notifications invox will refresh every time the tab is loaded.  
> >
> > ⋮ (_Snipped…_)  
> 
> This can prove to be problematic or disruptive to end-user workflows; for example:  
> 
> 1. A user may prefer not to dismiss an issue from their notifications inbox until after they've reviewed both it and a pull request which resolved it.  
> 1. A user may tap a link to a non-GitHub URI to oprn it in their browser or otherwise navigate away from the GitHub app to do something else, perhaps run a Web search for something if, say, they run into an unfamiliar term and want to look it up.  If this action outside the GitHub app takes long enough, the user may switch back to it only to find that it was suspended by the OS and reset back to the notifications inbox from whichever issue or PR they might've been looking at before.  With the old behavior, getting back to where you were is relatively easy:  you just tap the relevant issue or PR's notification inbox list entry again.  With the new behavior, you have to remember where you were before you switched away from the GitHub app and find it again by navigating through its associated repository's issue or PR list.  
> 
> Users should therefore be able to comtrol whether this new behavior triggers or not from the in-app setting page.  
> 
> (Feel free to merge this issue with #10276.)

**Comment by brianlovin** [comment]
Thanks for the feedback here! We're going to be following up on this change to adjust the behavior of the inbox refreshing so that threads won't get lost and you'll have the control you expect.
